x-pack/plugins/endpoint/common/models/event.ts

```diff
@@ -46,3 +47,42 @@ export function parentEntityId(event: ResolverEvent): string | undefined {
   }
   return event.process.parent?.entity_id;
 }
+
+export function eventCategoryDisplayName(event: ResolverEvent): EventCategory {
+  const eventTypeToNameMap = new Map<string, EventCategory | undefined>([
+    ['process', 'Process'],
+    ['alert', 'Alert'],
+    ['security', 'Security'],
+    ['file', 'File'],
+    ['network', 'Network'],
+    ['registry', 'Registry'],
+    ['dns', 'DNS'],
+    ['clr', 'CLR'],
+    ['image_load', 'Image Load'],
+    ['powershell', 'Powershell'],
+    ['wmi', 'WMI'],
+    ['api', 'API'],
+    ['user', 'User'],
+  ]);
+
+  // Returning "Process" as a catch-all here because it seems pretty general
+  let eventCategoryToReturn: EventCategory = 'Process';
+  if (isLegacyEvent(event)) {
+    const legacyFullType = event.endgame.event_type_full;
+    if (legacyFullType) {
+      const mappedLegacyCategory = eventTypeToNameMap.get(legacyFullType);
+      if (mappedLegacyCategory) {
+        eventCategoryToReturn = mappedLegacyCategory;
+      }
+    }
+  } else {
+    const eventCategories = event.event.category;
+    const eventCategory =
+      typeof eventCategories === 'string' ? eventCategories : eventCategories[0] || '';
```

Comment:
🎗️ Dangerous, but we have to make it fit into one of the known types until we take on the stuff Jonny was talking about w.r.t. allowing things to count twice.

Reply:
Yeah I'm thinking we'll probably do the mapping on the backend because we're going to need special logic there anyway to correctly display accurate stats for each type of event. My thought was that I'll return some stats that look something like:

```
{
  "dns": 5,
  "network": 10,
  "user": 100,
  ...
}
```

That way we don't have to duplicate the logic in the front end. I don't know what are your thoughts on keeping that logic in the backend?

Reply:
It all really depends on whether we _want_ to allow events to be "2 things": If we do, it makes sense to have the front-end aware of that concern. If not, it would be better abstracted away in the backend.

Reply:
> We're also going to need to pivot off of those statistics to e.g. _do something_ when a user clicks on "5 dns". I'm not sure how to handle the back-and-forth there; Send a list of `entity id`s or have an endpoint like `/related/byType/dns` and run it off of that

I think we could probably modify the related events endpoint to take a filter something like `/{id}/related?type=dns`

Reply:
> It all really depends on whether we want to allow events to be "2 things":

That's not really up to us though, it's up to ECS. Need to support 2+ going forward.

Reply:
Raised on https://github.com/elastic/endpoint-app-team/issues/379